### PR TITLE
chore: migrate to Angular v17

### PR DESCRIPTION
**Pull request by edbzn**
Angular v17 support

### CODE REVIEW

File: apps/ssr/server.ts.bak
Comment (vmasek) [inline apps/ssr/server.ts.bak:1]: I guess this backup file could be deleted as new one will be at use and "backup" will live in git history

File: libs/cdk/package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
     "url": "https://github.com/rx-angular/rx-angular.git"
   },
   "peerDependencies": {
-    "@angular/core": "^16.0.0",
+    "@angular/core": "^17.0.0",
```
Comment (lincolnthree) [inline libs/cdk/package.json:45]: These version ranges seem to cause issues when building against newer angular major versions.

Thoughts on making them a little less restrictive? E.g. `>=17.0.0`

```
npm WARN Could not resolve dependency:
npm WARN peer @angular/core@"^16.0.0" from @rx-angular/cdk@16.0.0
npm WARN node_modules/@rx-angular/cdk
npm WARN   @rx-angular/cdk@"^16.0.0" from the root project
npm WARN   1 more (@rx-angular/template)
```

Comment (hoebbelsB) [reply libs/cdk/package.json:45]: cc @LayZeeDK

Comment (edbzn) [reply libs/cdk/package.json:45]: It's the goal, we chose to sync RxA major version with Angular, this way we can guarantee compatibility, the problem with `>=17.0.0` is that we cannot guarantee compatibility for the next majors, it's a bet.

Comment (lincolnthree) [reply libs/cdk/package.json:45]: Makes sense. Thanks!

File: apps/ssr/server.ts [hunk 2]
```diff
@@ -1,44 +1,56 @@
...
-  const indexHtml = existsSync(join(distFolder, 'index.original.html')) ? 'index.original.html' : 'index';
+  const indexHtml = existsSync(join(distFolder, 'index.original.html'))
+    ? join(distFolder, 'index.original.html')
+    : join(distFolder, 'index.html');
 
-  // Our Universal express-engine (found @ https://github.com/angular/universal/tree/master/modules/express-engine)
-  server.engine('html', ngExpressEngine({
-    bootstrap: AppServerModule,
-  }));
+  const commonEngine = new CommonEngine();
```
Comment (BioPhoton) [inline apps/ssr/server.ts:18]: Nice Learned something!

Comment (eneajaho) [reply apps/ssr/server.ts:18]: And that's not all that changed. There's a lot more.

**Comment by eneajaho** [comment]
LGTM! Thanks  🎉

File: libs/cdk/package.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
...
     "rxjs": "^6.5.3 || ^7.4.0"
```
Comment (hoebbelsB) [inline libs/cdk/package.json:46]: is the rxjs version still valid? We changed some APIs like `connectable` as I remember correctly.

Comment (edbzn) [reply libs/cdk/package.json:46]: Yes, it's still valid, the PR where I changed some APIs is not merged yet (#1638), I'll check about compatibility in the corresponding PR.

**Pull request merged**